torchci/clickhouse_queries/queue_times_historical/query.sql

```diff
@@ -0,0 +1,28 @@
+SELECT
+    formatDateTime(
+            CASE
+                WHEN {granularity:String} = 'minute' THEN toStartOfMinute(q.time)
+                WHEN {granularity:String} = 'hour' THEN toStartOfHour(q.time)
+                WHEN {granularity:String} = 'day' THEN toStartOfDay(q.time)
+                WHEN {granularity:String} = 'week' THEN toStartOfWeek(q.time)
+                WHEN {granularity:String} = 'month' THEN toStartOfMonth(q.time)
+                WHEN {granularity:String} = 'year' THEN toStartOfYear(q.time)
+                ELSE toStartOfDay(q.time)  -- Default to day if granularity is not recognized
+                END,
+            '%Y-%m-%dT%H:%i:%s'
+    ) AS granularity_bucket,
+    /* misnomer, this is the max queue time, not the avg queue time */
+    avg(q.avg_queue_s) as avg_queue_s,
+    q.machine_type
+FROM
+    queue_times_historical q
+WHERE
+    q.time >= parseDateTimeBestEffort({startTime:String}) AND q.time < parseDateTimeBestEffort({stopTime:String})
```

Comment:
This goes with the previous comment, if we use `DateTime64` here, then the condition would just become `q.time >= {startTime: DateTime64(3)} AND q.time < {stopTime: DateTime64(3)}`

Reply:
There is no performance gain from this, but it simplifies the migration (the same set of parameters on the client work for Rockset and CH). I'd suggest we optimize when Rockset is gone.